File: 프로그래머스/lv1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.py
def solution(s):   
    answer = True
    num_p = num_y = 0       # s에 포함된 p와 y의 개수

    for a in s:
        if a == 'p' or a == 'P':
            num_p += 1
        elif a == 'y' or a == 'Y':
            num_y += 1
    
    if num_p != num_y:      # p와 y 개수가 다를 때만 False
        answer = False

    return answer

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int cntP = 0;
        int cntY = 0;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if ((c == 'p') || (c == 'P'))
                cntP++;
            if ((c == 'y') || (c == 'Y'))
                cntY++;
        }
        
        if (cntP != cntY)
            answer = false;

        return answer;
    }
}